mapper/pojo-standalone/src/main/java/org/hibernate/search/mapper/pojo/standalone/mapping/impl/StandalonePojoMapping.java

```diff
@@ -73,6 +80,15 @@ public class StandalonePojoMapping extends AbstractPojoMappingImplementor<Standa
 	@Override
 	public CompletableFuture<?> start(MappingStartContext context) {
 		integrationHandle = context.integrationHandle();
+		// check if indexing plan sync strategy was configured and cache it for further usage when creating sessions:
+		try ( BeanHolder<? extends IndexingPlanSynchronizationStrategy> beanHolder = INDEXING_PLAN_SYNCHRONIZATION_STRATEGY.getAndTransform(
+				context.configurationPropertySource(),
+				referenceOptional -> context.beanResolver().resolve( referenceOptional.orElse(
+						StandalonePojoMapperSettings.Defaults.INDEXING_PLAN_SYNCHRONIZATION_STRATEGY ) )
+		) ) {
+			this.configuredIndexingPlanSynchronizationStrategy = beanHolder.get();
```

Comment:
This is not an instance of `ConfiguredIndexingPlanSynchronizationStrategy`, though... ? I was suggesting to create a `ConfiguredIndexingPlanSynchronizationStrategy` and cache *that*, not the `IndexingPlanSynchronizationStrategy`.

Relatedly, note that extracting a bean from a `BeanHolder` and using it after the `BeanHolder` is closed is a big no-no, as CDI (for example) may execute some pre-destruct methods when `BeanHolder.close()` is called, rendering the bean unusable.

Reply:
Right... I've ended up caching the `IndexingPlanSynchronizationStrategy` as I thought - because the user can override the strategy at the session level it would somewhat duplicate that logic of creating the `ConfiguredIndexingPlanSynchronizationStrategy`. But I'll give it another shot. 
oh ok.. so we'd need to hold to the bean holder and push it to the closer in `StandalonePojoMapping#close` 👍🏻

Reply:
You don't necessarily need to duplicate the logic, or at least not a lot of it. IIRC I did it for the ORM mapper.